### DISCUSSION TITLE
How to add column in plugin

### DISCUSSION THREAD

**Discussion by Pascualsolutions** [wintercms/winter#1018] [General]
Hi again :)
I try to add some field (Seo and other) to plugin blog.i try to migrate with winter compouse:up /  update i try artisan migrate but cant find new column in database. Nothing to migrate is the msg.
Alex

**Comment by LukeTowers** [comment]
What have you done to provide the migration code to Winter? See https://wintercms.com/docs/v1.2/docs/plugin/updates for how migrations should be structured and reported to Winter. Please ensure your migrations are being done only in your plugin and that you're not modifying the code of the Winter.Blog plugin directly.

**Reply by Pascualsolutions** [reply]
I try to add field Meta_title and Meta_description etc. The plugin do not provide that field. I read documentation and i will try to migrate just plugin. Tnx for help :)

**Reply by LukeTowers** [reply]
If you're dealing with the [Winter.Blog plugin](https://github.com/wintercms/wn-blog-plugin) there is already a json column "metadata", so if you wanted to add additional data to the post records then you could refer to the docs ([extending forms](https://wintercms.com/docs/v1.2/docs/backend/forms#extending-form-behavior), [extending plugins](https://wintercms.com/docs/v1.2/docs/plugin/extending#extending-backend-forms))on how to extend the blog plugin's backend forms and just add in new fields storing their information under the metadata column instead:

```php
\Winter\Blog\Controllers\Posts::extendFormFields(function($form, $model, $context)
{
    if (!$model instanceof \Winter\Blog\Models\Post) {
        return;
    }

    $form->addTabFields([
        'metadata[my_seo]' => [
			'tab' => 'SEO',
			'type' => 'nestedform',
			'form' => [
				'fields' => [
					'meta_title' => [
						'label' => 'Meta Title',
						'span' => 'auto',
					],
					'meta_description' => [
						'label' => 'Meta Description',
						'span' => 'auto',
					],
				],
			],
        ],
    ]);
});
```

That way you don't have to touch the other plugin's database tables at all or deal with any migrations and it makes your plugin much more portable and play nicer with any other plugins that might extend the original plugin.

Additionally, have you taken a look at the [Winter.SEO](https://github.com/wintercms/wn-seo-plugin) plugin yet? It doesn't yet automatically add the fields for you (at least I don't think I added that yet) but it would be easy for it to do so, so if you'd like to use that plugin for it's other functionality let me know what you're looking for more specifically and I should be able to add it to the SEO plugin so that you don't even need to build your own plugin (for this functionality at least).
